### DISCUSSION TITLE
Data schema for `magnetic_state`

### DISCUSSION THREAD

**Discussion by JosePizarro3** [FAIRmat-NFDI/nomad-simulations#22] [Ideas]
@ndaelman-hu @JFRudzinski 

This is part of my notes done in #12 . I moved them here for consistency.
____

### Free atom `magnetic_moment`

The quantum numbers (L, S, J): depending on the relative size of the spin-orbit coupling, we need to use LS notation (small SOC) or JJ notation (large SOC). L, S are calculated from the individual electron ml and ms numbers as:

L = Σ<sub>i</sub> ml<sub>i</sub>
S = Σ<sub>i</sub> ms<sub>i</sub>
|L-S| <= J <= L+S

where i represents each electron in the `OrbitalState(n, l, ml, ms)`.

To calculate L and S we need to know the `OrbitalsState.occupation` for each orbital, using [Hund's rules](https://en.wikipedia.org/wiki/Hund%27s_rules):
1. Maximize S (first fill spin up 1/2, then spin down -1/2)
2. Maximize L (first fill larger ml and then smaller to negative)
3. Minimize SOC --> J=|L-S| if `AtomState` is less than half-full, J=L+S if `AtomState` is more than half-full.

For example, Mn+3 has a valence 3d4 `AtomState`:
![Capture](https://github.com/nomad-coe/nomad-schema-plugin-simulation-data/assets/112697669/82e502c8-ab38-4f16-970e-be234d976010)

The effective `magnetic_moment` is calculated using:
![image](https://github.com/nomad-coe/nomad-schema-plugin-simulation-data/assets/112697669/4400f59a-2a0c-4e31-b08b-59aff41ebe7d)

In this example, however, experimentally we find μ<sub>exp</sub>=4.82 μ<sub>B</sub> (instead of μ<sub>eff</sub>=0).

### The effect of the crystalline environment in the `magnetic_moment`

Due to the crystal environment, `OrbitalState` is feeling finite `crystal_field` / `onsite_energy` effects which depend on the symmetry of the crystal (e.g., perovskites break the d orbital symmetry into 3-fold degenerate t2g and 2-fold degenerate eg orbitals).

The [Slater-Koster TB method](https://journals.aps.org/pr/abstract/10.1103/PhysRev.94.1498) is a good starting point to obtain onsite energies.

In the example for Mn+3, there is an interesting effect playing role: orbital quenching (important in d-orbital systems). If instead of L=2, we consider L=0, we get μ<sub>eff</sub>=4.89 μ<sub>B</sub> (close to the experimental value). This happens when the crystal field effect is larger than SOC, breaking Hund's rules.

![Capture4](https://github.com/nomad-coe/nomad-schema-plugin-simulation-data/assets/112697669/cb8d7bdf-6c3d-40a5-8768-8c326d29ec6d)

Overall, we have 3 main energy scales which control the `OrbitalsState` which defines the `AtomsState.magnetic_moment`:
1. Crystal field
2. SOC
3. Hund's coupling (favors Hund's rule)
![Capture5](https://github.com/nomad-coe/nomad-schema-plugin-simulation-data/assets/112697669/2139c403-313e-4609-b8ba-3c85a6a24f27)
![Capture3](https://github.com/nomad-coe/nomad-schema-plugin-simulation-data/assets/112697669/2e926556-4cfa-4043-937e-d54dc0d84ce1)

___

## Idea for the schema

We need to:

1. Check `OrbitalsState.j_quantum_number` and `OrbitalsState.mj_quantum_number`. Are these necessary?
2. Define `AtomsState.L_total_quantum_number`, `AtomsState.S_total_quantum_number`, `AtomsState.J_total_quantum_number`. Here we could think whether we want to keep a string using `'(2S+1)LJ'` notation.
3. Define `AtomsState.magnetic_moment`, `AtomsState.spin_orbit_coupling` (this is, the pre-factor on the **L** · **S**).
4. Define `OrbitalsState.crystal_field` (or `OrbitalsState.onsite_energy`). Related with #11 
5. Is it possible to use Hund's rules + crystal_field + spin_orbit_coupling + `OrbitalsState.occupation` to obtain the quantum numbers L, S, J via normalization?
6. What about anisotropic spin Hamiltonians which go like H = D Sz^2 + E(Sx^2 - Sy^2)?
7. Via normalization using `AtomsState.magnetic_moment`, we can define `ModelSystem.magnetic_state` (or `ModelSystem.magnetic_order`): paramagnetic, ferromagnetic, antiferromagnetic (collinear or non-collinear, ferrimagnetic? altermagnetic?), spin glass.

We can also define `ModelSystem.magnetization`, which should be connected (perhaps, I am not 100% sure) with all the `AtomsState[i].magnetic_moment`. Finally, also add `ModelSystem.magnetic_state_q_vector` if possible.

___

## Note on where do we stop

I think we should not try to cover all potential cases in magnetism. For example, we can skip trying to characterize perfectly all the possible exchange interactions, and perhaps leave these for a near future.

**Comment by JosePizarro3** [comment]
@EBB2675

**Reply by EBB2675** [reply]
Thanks a lot! Let me make a draft ModelSystem.magnetization and ModelSystem.charge this week, that would go well for molecular systems, and we can discuss it there?
